Database/MongoDB/Transport/Tls.hs

```diff
@@ -0,0 +1,84 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-- | TLS connection to mongodb
+
+module Database.MongoDB.Transport.Tls
+(
+  connect,
+)
+where
+
+import Data.IORef
+import Data.Monoid
+import qualified Data.ByteString as ByteString
+import qualified Data.ByteString.Lazy as Lazy.ByteString
+import Data.Default.Class (def)
+import Control.Applicative ((<$>))
+import Control.Exception (bracketOnError)
+import Control.Monad (when, unless)
+import System.IO
+import Database.MongoDB (Pipe)
+import Database.MongoDB.Internal.Protocol (newPipeWith)
+import Database.MongoDB.Transport (Transport(Transport))
+import qualified Database.MongoDB.Transport as T
+import System.IO.Error (mkIOError, eofErrorType)
+import Network (connectTo, HostName, PortID)
+import qualified Network.TLS as TLS
+import qualified Network.TLS.Extra.Cipher as TLS
+import qualified Control.IO.Region as Region
+
+-- | Connect to mongodb using TLS
+connect :: HostName -> PortID -> IO Pipe
+connect host port = bracketOnError Region.open Region.close $ \r -> do
+  handle <- Region.alloc_ r
+    (connectTo host port)
+    hClose
+
+  let params = (TLS.defaultParamsClient host "")
+        { TLS.clientSupported = def
+            { TLS.supportedCiphers = TLS.ciphersuite_all}
+        , TLS.clientHooks = def
+            { TLS.onServerCertificate = \_ _ _ _ -> return []}
```

Comment:
I'm not sure it is a good idea to integrate the code as it is. I wrote this without security in mind, I just needed to connect to mongo instance which allows only ssl connection. I don't know tls enough, but I think here I disabled certificate validation.